### PR TITLE
fix javadoc and field name warnings

### PR DESCRIPTION
**Pull request by eristhea**
redo of #302 since I messed up a git merge

here's that with a clean diff!

### CODE REVIEW

File: mappings/net/minecraft/client/particle/GlowParticle.mapping [hunk 1]
```diff
@@ -12,7 +12,7 @@ CLASS net/minecraft/unmapped/C_ztzemsco net/minecraft/client/particle/GlowPartic
 		ARG 14 spriteProvider
 	CLASS C_abxthkae WaxOnFactory
 		FIELD f_cykyimlw spriteProvider Lnet/minecraft/unmapped/C_audtngae;
-		FIELD f_pighwgqn SPEED_FACTOR D
+		FIELD f_pighwgqn speedFactory D
```
Comment (IotaBread) [inline mappings/net/minecraft/client/particle/GlowParticle.mapping]:
```suggestion
		FIELD f_pighwgqn speedFactor D
```

File: mappings/net/minecraft/server/network/ServerLoginNetworkHandler.mapping [hunk 2]
```diff
@@ -16,7 +16,7 @@ CLASS net/minecraft/unmapped/C_rktsiyfy net/minecraft/server/network/ServerLogin
 	FIELD f_dosbbvgm loginTicks I
 	FIELD f_jtougkjq RANDOM Lnet/minecraft/unmapped/C_rlomrsco;
 	FIELD f_kankoaiw serverId Ljava/lang/String;
-	FIELD f_krozceue NEXT_AUTHENTICATOR_THREAD_ID Ljava/util/concurrent/atomic/AtomicInteger;
+	FIELD f_krozceue nextAuthenticatorThreadId Ljava/util/concurrent/atomic/AtomicInteger;
```
Comment (IotaBread) [inline mappings/net/minecraft/server/network/ServerLoginNetworkHandler.mapping:19]: We use upper snake-case for atomics iirc?

**Review by IotaBread — COMMENTED** [review]
I think most of these were named using upper snake-casing because their values are constant (i.e. `public final float VALUE = 3.2F;`), so we should update the linter to catch these properly

File: mappings/net/minecraft/server/network/ServerLoginNetworkHandler.mapping (hunk 2 above)
Comment (eristhea) [reply mappings/net/minecraft/server/network/ServerLoginNetworkHandler.mapping:19]: `FieldNamingChecker`, line 27-31
```java
        if (isAtomic) {
            if (startsWithUppercase(mapping.targetName())) {
                errorReporter.warning("atomic field starts with uppercase character '" + mapping.targetName().charAt(0) + "'");
            }
        }
```

Comment (OroArmor) [reply mappings/net/minecraft/server/network/ServerLoginNetworkHandler.mapping:19]: Yeah i think this should move away from SNAKE_CASE

**Review by OroArmor — APPROVED** [review]

**Review by IotaBread — APPROVED** [review]

**Review by NoComment1105 — APPROVED** [review]

**Pull request merged**